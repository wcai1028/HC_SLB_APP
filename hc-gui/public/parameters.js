var parameters = {
  // 'BASE_URL': 'https://10.0.1.58:8443/api/v2',
  BASE_URL: 'https://10.0.1.58:8443/api/v2',
  CLOUD_ACCOUNT_ID: '339858680748',
  DEPLOYMENT_TYPE: 'SaaS', // <onprem/SaaS>, SaaS = AWS,Azure,Openstack   onprem=vmwarer
  WHITELABELED_PROVIDER: [''],
  EDUCATION_LINK: 'http://docs.lc.a10networks.com/4.0.0/',
  TDM_URL: 'https://107.178.208.19:7443',
  VERSION_NUMBER: '4.0.1',
  BUILD_NUMBER: '78755630314bbbbe41122885ae874ed76b4aac3a',
  SUPPORT_URL: 'http://www.a10networks.com/cloud-support',
}
