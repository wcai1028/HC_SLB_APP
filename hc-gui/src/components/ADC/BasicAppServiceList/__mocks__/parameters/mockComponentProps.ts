
export default (isLoading: boolean = false) => {
  if (isLoading) {
    return {
      data: new Array,
      isLoading: true,
    }
  } else {
    return {
      data: [
        {
          'app-svc': 'appSvc1',
          vport: '80+tcp',
          'app-svc-group': 'appSvcGrp1',
        },
        {
          'app-svc': 'appSvc2',
          vport: '81+udp',
          'app-svc-group': 'appSvcGrp2',
        },
      ],
      isLoading: false,
    }
  }

}
