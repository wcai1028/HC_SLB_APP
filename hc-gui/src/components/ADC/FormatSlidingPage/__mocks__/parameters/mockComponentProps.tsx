import * as React from 'react'

const minProps = {
  isOpen: true,
  children: <p>Inner Component</p>,
}

const commonProps = {
  isOpen: true,
  title: 'Common Unit Test',
  description: 'This is a common Unit Test',
  children: <p>Inner Component</p>,
  showBack: false,
  saveText: null,
  disableSave: false,
  hideCancel: false,
  customizedButtons: null,
}

const specialProps = {
  isOpen: true,
  title: 'Special Unit Test',
  description: 'This is a special Unit Test',
  children: <p>Inner Component</p>,
  showBack: true,
  saveText: 'TestSave',
  disableSave: true,
  hideCancel: true,
  customizedButtons: <button>Customized</button>,
}

export default (isSpecial?: boolean, callbackOK?: any, callbackCancel?: any) => {
  if (isSpecial === undefined && callbackOK === undefined && callbackCancel === undefined) {
    return minProps
  }

  if (isSpecial) {
    return {
      ...specialProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  } else {
    return {
      ...commonProps,
      onRequestOk: callbackOK,
      onRequestClose: callbackCancel,
    }
  }
}
