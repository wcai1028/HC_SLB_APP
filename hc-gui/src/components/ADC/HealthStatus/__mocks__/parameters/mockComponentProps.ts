export default {
  headerProps: {
    type: 'header',
  },
  ongoingProps: {
    type: 'ongoing',
  },
  lowwarnProps: {
    type: 'lowwarn',
  },
  warningProps: {
    type: 'warning',
  },
  stoppedProps: {
    type: 'stopped',
  },
  infoProps: {
    text: 'test info',
    type: 'info',
    tooltip: 'test tooltip',
  },
  unusedProps: {
    type: 'ununsed',
    default: 'test default'
  },
  numberProps: {
    type: 'number',
    size: 32,
  },
  undefinedProps: {
  },
}
