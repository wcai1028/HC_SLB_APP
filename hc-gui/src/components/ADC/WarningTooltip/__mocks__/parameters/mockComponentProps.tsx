import * as React from 'react'

interface ITestDivProps {
  onClick: () => void,
}

class TestDiv extends React.Component<ITestDivProps> {

  handleClick = () => {
    const { onClick } = this.props
    if (onClick) {
      onClick()
    }
  }

  render() {
    return <div onClick={this.handleClick}>{this.props.children}</div>
  }
}

export const mockWarningTooltipProps =  (useDefault: boolean, onConfirm: any, onVisibleChange: any) => {
  if (useDefault) {
    return {
      title: '',
      description: [],
      action: '',
      onConfirm,
      content: null,
      arrowPointAtCenter: null,
      placement: null,
      trigger: null,
      visible: undefined,
      onVisibleChange: null,
    }
  } else {
    return {
      title: 'Test Title',
      description: ['Test description1', 'Test descriptions2'],
      action: 'Save',
      onConfirm,
      content: (
        title: string,
        description: string[],
        action: string,
        onConfirm: () => void,
      ) => {
        return <>
          <div>{title}</div>
          <div>
            {description.map((data: string, index: number) => {
              return <span key={index}>{data}</span>
            })}
          </div>
          <TestDiv onClick={onConfirm} >{action}</TestDiv>
        </>
      },
      arrowPointAtCenter: false,
      placement: 'bottomRight',
      trigger: 'click',
      visible: false,
      onVisibleChange,
    }
  }
}

export const mockInfoTooltipProps = (isEmpty: boolean, isDefault: boolean, isTest: boolean) => {
  if (isEmpty) {
    return {}
  } else if (isDefault) {
    return {
      objList: [
        { value: 'Test value1' },
        { value: 'Test value2' },
        { value: 'Test value3' },
      ],
    }
  } else {
    if (isTest === true) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: null,
        textDisplay: 'Test text',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    } else if (isTest === false) {
      return {
        objList: [
          { value: 'Test value1' },
          { value: 'Test value2' },
          { value: 'Test value3' },
        ],
        componentDisplay: <div>Test component</div>,
        textDisplay: '',
        renderText: (obj: IObject) => {
          return <div>{obj.value}</div>
        },
        actions: 'Test action',
      }
    }
    return {}
  }
}
