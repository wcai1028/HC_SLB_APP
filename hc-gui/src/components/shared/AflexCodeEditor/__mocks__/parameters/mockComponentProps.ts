
export default (onAFlexChange: any, hasDiff: boolean = false, isBeforeStyle: boolean = false) => {
  if (hasDiff) {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: [
        ['replace', 1, 3, 5, 7],
        ['insert', 1, 3, 5, 7],
        ['delete', 1, 3, 5, 7],
        ['others', 1, 3, 5, 7],
      ],
      version: isBeforeStyle ? 'before' : 'after',
      id: '123456',
    }
  } else {
    return {
      onAFlexChange,
      value: 'test-value',
      options: {
        readOnly: true,
      },
      diff: null,
      version: 'unit_test',
      id: '123456',
    }
  }

}
