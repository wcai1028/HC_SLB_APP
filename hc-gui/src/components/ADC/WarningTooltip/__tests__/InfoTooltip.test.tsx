import * as React from 'react'
import * as ReactTestRenderer from 'react-test-renderer'

import { mockInfoTooltipProps } from '../__mocks__/parameters/mockComponentProps'

const renderTestComponent = (props: any) => {
  const HCProvider = require('src/HCProvider').default
  const InfoTooltip = require('../InfoTooltip').default

  return ReactTestRenderer.create(
    <HCProvider>
      <InfoTooltip {...props} />
    </HCProvider>
  )
}

describe('Unit Test: InfoTooltip ', () => {

  describe('should render properly: ', () => {
    test('should render properly with empty props', (done: any) => {
      const props = mockInfoTooltipProps(true, false, false)
      const testComponent = renderTestComponent(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with default props', (done: any) => {
      const props = mockInfoTooltipProps(false, true, false)
      const testComponent = renderTestComponent(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with text', (done: any) => {
      const props = mockInfoTooltipProps(false, false, true)
      const testComponent = renderTestComponent(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

    test('should render properly with component', (done: any) => {
      const props = mockInfoTooltipProps(false, false, false)
      const testComponent = renderTestComponent(props)

      setImmediate(() => {
        expect(testComponent.toJSON()).toMatchSnapshot()
        done()
      })
    })

  })

})
