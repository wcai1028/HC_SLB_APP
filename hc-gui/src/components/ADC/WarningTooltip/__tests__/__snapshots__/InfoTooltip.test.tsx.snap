// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test: InfoTooltip  should render properly:  should render properly with component 1`] = `
<div
  className="mock-a10-tooltip"
>
  <div>
    Test component
  </div>
  <div
    className="dirtyList"
  >
    <ul
      className="description"
    >
      <li>
        <div>
          Test value1
        </div>
      </li>
      <li>
        <div>
          Test value2
        </div>
      </li>
      <li>
        <div>
          Test value3
        </div>
      </li>
      <div
        className="action"
      >
        Test action
      </div>
    </ul>
  </div>
</div>
`;

exports[`Unit Test: InfoTooltip  should render properly:  should render properly with default props 1`] = `
<div
  className="mock-a10-tooltip"
>
  <a>
    3
  </a>
  <div
    className="dirtyList"
  >
    <ul
      className="description"
    >
      <li>
         
      </li>
      <li>
         
      </li>
      <li>
         
      </li>
    </ul>
  </div>
</div>
`;

exports[`Unit Test: InfoTooltip  should render properly:  should render properly with empty props 1`] = `"0"`;

exports[`Unit Test: InfoTooltip  should render properly:  should render properly with text 1`] = `
<div
  className="mock-a10-tooltip"
>
  <a>
    Test text
  </a>
  <div
    className="dirtyList"
  >
    <ul
      className="description"
    >
      <li>
        <div>
          Test value1
        </div>
      </li>
      <li>
        <div>
          Test value2
        </div>
      </li>
      <li>
        <div>
          Test value3
        </div>
      </li>
      <div
        className="action"
      >
        Test action
      </div>
    </ul>
  </div>
</div>
`;
